test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -150,17 +150,32 @@ class test_ucp_sockaddr : public ucp_test {
                 ucs::is_inet_addr(ifa->ifa_addr) &&
                 ucs::is_rdmacm_netdev(ifa->ifa_name))
             {
+                saddrs.push_back(ucs::sock_addr_storage());
                 status = ucs_sockaddr_sizeof(ifa->ifa_addr, &size);
                 ASSERT_UCS_OK(status);
-                m_test_addr.set_sock_addr(*ifa->ifa_addr, size);
-                m_test_addr.set_port(0); /* listen on any port then update */
-
-                freeifaddrs(ifaddrs);
-                return;
+                saddrs.back().set_sock_addr(*ifa->ifa_addr, size);
+                saddrs.back().set_port(0); /* listen on any port then update */
             }
         }
+
         freeifaddrs(ifaddrs);
-        UCS_TEST_SKIP_R("No interface for testing");
+
+        if (saddrs.empty()) {
+            UCS_TEST_SKIP_R("No interface for testing");
+        }
+
+        static const std::string ud_or_dc_tls[] = { "ud", "ud_v", "ud_x",
+                                                    "dc", "dc_x", "ib" };
+
+        bool has_dc_or_ud = has_any_transport(
+                        std::vector<std::string>(ud_or_dc_tls,
+                                                 ud_or_dc_tls +
+                                                 ucs_array_size(ud_or_dc_tls)));
+
+        /* randomize iface excepting UD and DC transports which have a bug
+           on transport level, PR #5237 fixes ud/udx */
```

Comment:
FIXME select random interface, except for  UD and DC transports, which do not yet support having different gid_index for different UCT endpoints on same iface